### ISSUE TITLE
Autodetect initializers

### ISSUE THREAD

**Issue by charlieMonroe** [drkameleon/iswift#91]
This is related to the `[NSURL URLWithString:]` issue. Here is another example:

```
XURegex *regex = [XURegex regexWithPattern:@"\\{\\s*'label'\\s*:\\s*'(?P<H>\\d+)p',\\s*'file'\\s*:\\s*'(?P<URL>[^']+)'" andOptions:XURegexOptionCaseless];
```

This currently gets translated to:

```
let regex = XURegex.regexWithPattern("\\{\\s*'label'\\s*:\\s*'(?P<H>\\d+)p',\\s*'file'\\s*:\\s*'(?P<URL>[^']+)'", andOptions: XURegexOptionCaseless)
```

While it should be:

```
let regex = XURegex(pattern: "\\{\\s*'label'\\s*:\\s*'(?P<H>\\d+)p',\\s*'file'\\s*:\\s*'(?P<URL>[^']+)'", andOptions: .Caseless)
```

Generally, if the class' name without the first two/three letters is the prefix of a class method that continues `With`, it should be considered an initializer.

Other examples:

```
[XUQualityDownload qualityDownloadWithURL:URLString andHeight:512];
[NSCharacterSet characterSetWithCharactersInString:@"abcd"];
```

Currently are:

```
XUQualityDownload.qualityDownloadWithURL(URLString, andHeight: 512)
NSCharacterSet(charactersetwithcharactersinstring: "abcd") // Here it "kinda works"
```

Should be:

```
XUQualityDownload(URL: URLString, andHeight: 512)
NSCharacterSet(charactersInString: "abcd")
```

**Comment by drkameleon** [comment]
Related: https://github.com/drkameleon/iswift/issues/64

**Comment by drkameleon** [comment]
Well... that one was a bit tricky (since we have to deal with lots of subtle variations), but I think I made it.

**Example:**

```objective-c
@class XURegex;
@class XUQualityDownload;

@implementation aClass 

-(void)someFunc {

    NSURL* u = [NSURL URLWithString:@"SomeString"];
    NSString* s = [NSString stringWithFormat:@"Some Format: %@", @"Done"];
    XURegex *regex = [XURegex regexWithPattern:@"\\{\\s*'label'\\s*:\\s*'(?P<H>\\d+)p',\\s*'file'\\s*:\\s*'(?P<URL>[^']+)'" andOptions:XURegexOptionCaseless];
    XUQualityDownload* qd = [XUQualityDownload qualityDownloadWithURL:URLString andHeight:512];
    NSCharacterSet* cs = [NSCharacterSet characterSetWithCharactersInString:@"abcd"];
}

@end
```

**Output:**

```swift
class aClass {

	func someFunc() {
		var u: NSURL = NSURL(string: "SomeString")
		var s: String = "Some Format: \("Done")"
		var regex: XURegex = XURegex(pattern: "\\{\\s*'label'\\s*:\\s*'(?P<H>\\d+)p',\\s*'file'\\s*:\\s*'(?P<URL>[^']+)'", andOptions: XURegexOptionCaseless)
		var qd: XUQualityDownload = XUQualityDownload(URL: URLString, andHeight: 512)
		var cs: NSCharacterSet = NSCharacterSet(charactersInString: "abcd")
	
	}
	
}
```

------

So,... issue solved as of the upcoming 2.0. :)
